File: templates/nodejs/function.js
const express = require("express");
const app = express();

app.use(express.json());
app.use(express.text());

app.all("/", (req, res) => {
    res.json({
        message: "Hello from Node.js function!",
        method: req.method,
        path: req.path,
        headers: req.headers,
        body: req.body,
    });
});

app.listen(8080, "0.0.0.0", () => {
    console.log("Node.js function server running on port 8080");
});

File: templates/python/function.py
from flask import Flask, request, jsonify
import json

app = Flask(__name__)


@app.route("/", methods=["GET", "POST", "PUT", "DELETE"])
def handler():
    return jsonify(
        {
            "message": "Hello from Python function!",
            "method": request.method,
            "path": request.path,
            "headers": dict(request.headers),
            "body": request.get_data().decode("utf-8") if request.get_data() else None,
        }
    )


if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8080)
